packages/charts/react-charts/library/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -162,50 +221,74 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
-      if (plotlyInput.data[0].mode === 'markers') {
-        throw new Error(`Unsupported chart - type :${plotlyInput.data[0]?.type}, mode: ${plotlyInput.data[0]?.mode}`);
-      }
-      const isAreaChart = plotlyInput.data.some(
-        (series: PlotData) => series.fill === 'tonexty' || series.fill === 'tozeroy',
+  switch (chart.type) {
+    case 'donut':
+      return (
+        <DonutChart
+          {...transformPlotlyJsonToDonutProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
+    case 'horizontalbar':
+      return (
+        <HorizontalBarChartWithAxis
+          {...transformPlotlyJsonToHorizontalBarWithAxisProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
+    case 'groupedverticalbar':
+      return (
+        <GroupedVerticalBarChart
+          {...transformPlotlyJsonToGVBCProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
+    case 'verticalstackedbar':
+      return (
+        <VerticalStackedBarChart
+          {...transformPlotlyJsonToVSBCProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
       );
-      const renderChartJsx = (chartProps: LineChartProps) => {
-        if (isAreaChart) {
-          throw new Error(
-            `Unsupported chart type :${plotlyInput.data[0]?.type}, fill: ${
-              (plotlyInput.data[0] as Partial<PlotData>)?.fill
-            }`,
-          );
-        }
-        return <LineChart {...chartProps} />;
-      };
-      return checkAndRenderChart(renderChartJsx, isAreaChart);
     case 'heatmap':
-      throw new Error(`Unsupported chart type :${plotlyInput.data[0]?.type}`);
+      return (
+        <HeatMapChart
+          {...transformPlotlyJsonToHeatmapProps(plotlyInputWithValidData)}
+          {...commonProps}
+          legendProps={{}}
+        />
+      );
     case 'sankey':
-      throw new Error(`Unsupported chart type :${plotlyInput.data[0]?.type}`);
-    case 'indicator':
+      return (
+        <SankeyChart
+          {...transformPlotlyJsonToSankeyProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
     case 'gauge':
-      throw new Error(`Unsupported chart - type: ${plotlyInput.data[0]?.type}, mode: ${plotlyInput.data[0]?.mode}`);
-    case 'histogram':
       return (
-        <VerticalBarChart {...transformPlotlyJsonToVBCProps(plotlySchema, colorMap, isDarkTheme)} {...commonProps} />
+        <GaugeChart
+          {...transformPlotlyJsonToGaugeProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
+    case 'verticalbar':
+      return (
+        <VerticalBarChart
+          {...transformPlotlyJsonToVBCProps(plotlyInputWithValidData, colorMap, isDarkTheme)}
+          {...commonProps}
+        />
+      );
+    case 'area':
+    case 'line':
```

Comment:
>line [](http://example.com/codeflow?start=10&length=4)

ToDo: now that we have scatter as separate chart, we need scatter as a new chart type. Also need to handle it properly in v8 declarative chart

Reply:
marked this as a TODO and will follow up on it later